### PR TITLE
Fix __title__ and doxygen \addtogroup blocks

### PR DESCRIPTION
**Pull request by luzpaz**
Still needs review

### CODE REVIEW

File: src/Mod/Fem/femsolver/elmer/equations/linear.py [hunk 1]
```diff
@@ -20,11 +20,12 @@
 # *                                                                         *
 # ***************************************************************************
 
-
-__title__ = "_Linear"
+__title__ = "FreeCAD FEM _Linear"
```
Comment (luzpaz) [inline src/Mod/Fem/femsolver/elmer/equations/linear.py]: should I remove the `_` from `_Linear` ?

File: src/Mod/Fem/femsolver/elmer/equations/nonlinear.py [hunk 2]
```diff
@@ -20,11 +20,12 @@
 # *                                                                         *
 # ***************************************************************************
 
-
-__title__ = "_NonLinear"
+__title__ = "FreeCAD FEM _NonLinear"
```
Comment (luzpaz) [inline src/Mod/Fem/femsolver/elmer/equations/nonlinear.py]: Same question, remove `_` ?

File: src/Mod/Fem/femsolver/elmer/sifio.py [hunk 3]
```diff
@@ -20,6 +20,12 @@
 # *                                                                         *
 # ***************************************************************************
 
+__title__ = "FreeCAD FEM ???"
```
Comment (luzpaz) [inline src/Mod/Fem/femsolver/elmer/sifio.py]: What is the correct `__title__`

File: src/Mod/Fem/femsolver/elmer/writer.py [hunk 4]
```diff
@@ -20,11 +20,12 @@
 # *                                                                         *
 # ***************************************************************************
 
-
-__title__ = "FemWriterElmer"
+__title__ = "FreeCAD FEM FemWriterElmer"
```
Comment (luzpaz) [inline src/Mod/Fem/femsolver/elmer/writer.py]: Please check the name

File: src/Mod/Fem/femsolver/equationbase.py [hunk 5]
```diff
@@ -20,11 +20,12 @@
 # *                                                                         *
 # ***************************************************************************
 
-
-__title__ = "_Base"
+__title__ = "FreeCAD FEM _Base"
```
Comment (luzpaz) [inline src/Mod/Fem/femsolver/equationbase.py]: remove the `_` ?

File: src/Mod/Fem/femsolver/reportdialog.py [hunk 6]
```diff
@@ -20,11 +20,12 @@
 # *                                                                         *
 # ***************************************************************************
 
-
-__title__ = "reportdialog"
+__title__ = "FreeCAD FEM reportdialog"
```
Comment (luzpaz) [inline src/Mod/Fem/femsolver/reportdialog.py]: `reportdialog` is ok? Or can it be `Report Dialog`?

File: src/Mod/Fem/femsolver/run.py [hunk 7]
```diff
@@ -20,11 +20,12 @@
 # *                                                                         *
 # ***************************************************************************
 
-
-__title__ = "run"
+__title__ = "FreeCAD FEM run"
```
Comment (luzpaz) [inline src/Mod/Fem/femsolver/run.py]: `run` what ?

File: src/Mod/Fem/femsolver/settings.py [hunk 8]
```diff
@@ -20,11 +20,12 @@
 # *                                                                         *
 # ***************************************************************************
 
-
-__title__ = "settings"
+__title__ = "FreeCAD FEM settings"
```
Comment (luzpaz) [inline src/Mod/Fem/femsolver/settings.py]: is `settings` correct? or can there be more elaboration ?

File: src/Mod/Fem/femsolver/signal.py [hunk 9]
```diff
@@ -20,12 +20,15 @@
 # *                                                                         *
 # ***************************************************************************
 
-
-__title__ = "signal"
+__title__ = "FreeCAD FEM signal"
```
Comment (luzpaz) [inline src/Mod/Fem/femsolver/signal.py]: is `signal` enough?

File: src/Mod/Fem/femsolver/task.py [hunk 10]
```diff
@@ -20,11 +20,12 @@
 # *                                                                         *
 # ***************************************************************************
 
-
-__title__ = "task"
+__title__ = "FreeCAD FEM task"
```
Comment (luzpaz) [inline src/Mod/Fem/femsolver/task.py]: what `task` ?

Comment (berndhahnebach) [reply src/Mod/Fem/femsolver/task.py]: FreeCAD FEM solver task

File: src/Mod/Fem/femsolver/signal.py (hunk 9 above)
Comment (berndhahnebach) [reply src/Mod/Fem/femsolver/signal.py]: FreeCAD FEM solver signal

File: src/Mod/Fem/femsolver/settings.py (hunk 8 above)
Comment (berndhahnebach) [reply src/Mod/Fem/femsolver/settings.py]: FreeCAD FEM solver settings

File: src/Mod/Fem/femsolver/run.py (hunk 7 above)
Comment (berndhahnebach) [reply src/Mod/Fem/femsolver/run.py]: FreeCAD FEM solver run

File: src/Mod/Fem/femsolver/reportdialog.py (hunk 6 above)
Comment (berndhahnebach) [reply src/Mod/Fem/femsolver/reportdialog.py]: FreeCAD FEM solver report dialog

File: src/Mod/Fem/femsolver/equationbase.py (hunk 5 above)
Comment (berndhahnebach) [reply src/Mod/Fem/femsolver/equationbase.py]: FreeCAD FEM solver elmer equation _Base

File: src/Mod/Fem/femsolver/elmer/writer.py (hunk 4 above)
Comment (berndhahnebach) [reply src/Mod/Fem/femsolver/elmer/writer.py]: FreeCAD FEM solver elmer writer

File: src/Mod/Fem/femsolver/elmer/sifio.py (hunk 3 above)
Comment (berndhahnebach) [reply src/Mod/Fem/femsolver/elmer/sifio.py]: FreeCAD FEM solver elmer sifio

File: src/Mod/Fem/femsolver/elmer/equations/nonlinear.py (hunk 2 above)
Comment (berndhahnebach) [reply src/Mod/Fem/femsolver/elmer/equations/nonlinear.py]: FreeCAD FEM solver elmer equation _NonLinear

File: src/Mod/Fem/femsolver/elmer/equations/linear.py (hunk 1 above)
Comment (berndhahnebach) [reply src/Mod/Fem/femsolver/elmer/equations/linear.py]: FreeCAD FEM solver elmer equation _Linear

**Comment by luzpaz** [comment]
@berndhahnebach 
I resolved all the fixes you oriented me in -> 8d7dae0
I also found some other non-uniform things and fixed + added Fenics -> b487187

**Comment by berndhahnebach** [comment]
I had a look at all modules in femsolver package. How about this?

See https://github.com/berndhahnebach/FreeCAD_bhb/commit/81be787523d20ed1828e3893ceb413597d0ab100

It is on top of your commits. If it fits for you too we can got for the other fem packages.

bernd

**Comment by luzpaz** [comment]
FYI, since we're tackling the FEM header files, I figured we'd also make the copyright and developer info uniform as well. In eb28d5f you can see I just did it for `femsolver/` so far. 

I'm doing the same thing with [TechDraw WB](https://github.com/FreeCAD/FreeCAD/compare/master...WandererFan:TD-Header-Uniformity). I'm using the pattern that Werner uses when he creates a new page.
`Copyright (c) YEAR DevFirstName DevLastName <email@address.com>` 

Are you OK with that ?

**Comment by luzpaz** [comment]
Sorry, @berndhahnebach but I didn't know how to give this latest commit a16ef0f as much love. Can you jump in. Im fact, I think you can actually manually make changes to the commit itself and push changes to it (since you're the admin of the repo)

**Comment by berndhahnebach** [comment]
> I'm doing the same thing with [TechDraw WB](https://github.com/FreeCAD/FreeCAD/compare/master...WandererFan:TD-Header-Uniformity). I'm using the pattern that Werner uses when he creates a new page.
> `Copyright (c) YEAR DevFirstName DevLastName <email@address.com>`
> 
> Are you OK with that ?

of course!   It is what werner does and it is what is in the template of FreeCAD master: https://github.com/FreeCAD/FreeCAD/blob/master/src/Tools/_TEMPLATE_/App/PreCompiled.cpp#L2

cheers bernd

**Comment by berndhahnebach** [comment]
https://github.com/berndhahnebach/FreeCAD_bhb/commit/6633ee1419344ce54649d2e23c2246522987cccd

**Comment by berndhahnebach** [comment]
either we fill ALL __init.py__ or we clear all of them ... I would clear them, how about you?

**Comment by luzpaz** [comment]
> either we fill ALL **init.py** or we clear all of them ... I would clear them, how about you?

I'm cool with clearing them :+1:

**Comment by berndhahnebach** [comment]
What's next?

**Comment by luzpaz** [comment]
what branch do i rebase this now, `femtemp` or `femdev` ?

**Comment by berndhahnebach** [comment]
best is to use femdev. femtmp is the branch I play with and cherry-pick all kind of code, and delete commtis and and and ...

**Comment by berndhahnebach** [comment]
should I cheery-pick your commits and my into femdev for you ?

**Comment by luzpaz** [comment]
no need. I will take care of it.

**Comment by berndhahnebach** [comment]
I added another two code formating commits ...

https://github.com/berndhahnebach/FreeCAD_bhb/commit/cb0a97ae4e2ece0e95e04e69ca84c691b5e36c2e
https://github.com/berndhahnebach/FreeCAD_bhb/commit/e9cfa953e4b8c65126c1f00528d47318e782adef

Would you like to continue or should I go for a PR?

**Comment by luzpaz** [comment]
@berndhahnebach i think you should go to PR. The process is getting a little too murky. I think we made good progress and we can continue when needed. This will give me the opportunity to cleanup my git environment a bit (i have way too many stashed commits that I need to sort out). So green light from me.

**Comment by berndhahnebach** [comment]
great work :-)

https://github.com/FreeCAD/FreeCAD/pull/1799

**Comment by luzpaz** [comment]
after it's merged, can you please rebase femdev ? Thanks!

**Comment by berndhahnebach** [comment]
FYI: Python does not like to ü in Jürgen ... https://github.com/FreeCAD/FreeCAD/commit/86455527ddb9eea15f5ae562c2f02b921d4c1a11

**Comment by berndhahnebach** [comment]
https://github.com/berndhahnebach/FreeCAD_bhb/commits/femdev

**Pull request closed without merging**